src/verify_expr.ml

```diff
@@ -2049,32 +2105,63 @@ module VerifyExpr(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
           check_call targs h args $. fun h env retval ->
           cont (chunk::h) env retval
       end
-    | NewObject (l, cn, args) ->
+    | NewObject (l, cn, args, targs) ->
       inheritance_check cn l;
       if pure then static_error l "Object creation is not allowed in a pure context" None;
-      let {cctors} = List.assoc cn classmap in
+      let {cctors; ctpenv} = List.assoc cn classmap in
       let args' = List.map (fun e -> check_expr (pn,ilist) tparams tenv e) args in
       let argtps = List.map snd args' in
-      let consmap' = List.filter (fun (sign, _) -> is_assignable_to_sign (Some pure) argtps sign) cctors in
+      let argtps_erased = List.map erase_type argtps in
+      let consmap' = 
+        match targs with 
+          (* The new object is a parameterised type *)
+          Some(targs) -> 
+            (* No support for the diamond notation yet, so we always have the types explicitly provided *)
+            let consWithErasedSigns = List.map (fun (sign,cinfo) -> (*Iterate ovr constructors*)
+            (List.map erase_type sign, cinfo)) cctors 

```

Comment:
Fix typo and indentation. Ook: geen haakjes over meerdere regels; gebruik dan `begin` ... `end`.